_posts/2018-08-23-debos-in-docker-the-second-attempt.md

```diff
@@ -0,0 +1,139 @@
+2018/08/21 19:17:06 debootstrap.log | gpgv: Signature made Tue Aug 21 14:20:20 2018 UTC
+2018/08/21 19:17:06 debootstrap.log | gpgv:                using RSA key A1BD8E9D78F7FE5C3E65D8AF8B48AD6246925553
+2018/08/21 19:17:06 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (7.0/wheezy) <ftpmaster@debian.org>"
+2018/08/21 19:17:06 debootstrap.log | gpgv: Signature made Tue Aug 21 14:20:20 2018 UTC
+2018/08/21 19:17:06 debootstrap.log | gpgv:                using RSA key 126C0D24BD8A2942CC7DF8AC7638D0442B90D010
+2018/08/21 19:17:06 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (8/jessie) <ftpmaster@debian.org>"
+2018/08/21 19:17:06 debootstrap.log | gpgv: Signature made Tue Aug 21 14:20:20 2018 UTC
+2018/08/21 19:17:06 debootstrap.log | gpgv:                using RSA key 16E90B3FDF65EDE3AA7F323C04EE7237B7D453EC
+2018/08/21 19:17:06 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>"
+2018/08/21 19:17:06 debootstrap.log | mount: /proc: permission denied.
+2018/08/21 19:17:06 Action `debootstrap` failed at stage Run, error: exit status 1
+```
+
+To overcome this `/proc` access issue I have decided to run the container in
+the
+[privileged mode](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities).
+Generally, I try to avoid doing that whenever not strictly required. Scoping
+the actual access requirements and reducing the container access to the host
+system might be another task to look at. For now, I'm sticking with the
+`privileged` mode enabled without much deeper analysis.
+
+## One more issue
+
+One more issue to solve was:
+
+```
+2018/08/21 14:17:53 debootstrap.log | Processing triggers for libc-bin (2.27-5) ...
+2018/08/21 14:17:53 debootstrap.log | /debootstrap/debootstrap: 1363: /debootstrap/debootstrap: cannot open //var/lib/apt/lists/debootstrap.invalid_dists_buster_main|contrib|no
+n-free_binary-arm64_Packages: No such file
+2018/08/21 14:17:53 Action `debootstrap` failed at stage Run, error: exit status 2
+Powering off.
+```
+
+It appears that there is a
+[debootstrap bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=806780)
+hanging in the `Debian` bug tracker since the `Dec 2015`. The issue
+[was resolved](https://salsa.debian.org/installer-team/debootstrap/commit/792ab830a892ccfaaca156eace00172d3432023a)
+in the `debootstrap`  `1.0.96`. Unfortunately, the `debootstrap` version shipped
+in my `Debian Stretch` container is `1.0.89`. As a quick solution, I decided to
+provide the patch manually:
+
+```
+COPY files/0001-Fix-multiple-components-usage-for-foreign.patch /usr/share/debootstrap
+RUN cd /usr/share/debootstrap && \
+    patch < 0001-Fix-multiple-components-usage-for-foreign.patch && \
+    rm /usr/share/debootstrap/0001-Fix-multiple-components-usage-for-foreign.patch && \
+    apt-get autoremove -y patch && \
+    rm -rf /var/lib/apt/lists/*
+```
```

Comment:
@macpijan alternative can be [stretch-backports repo](https://packages.debian.org/stretch-backports/debootstrap) - I wrote about it in my new blog post. Backports are clean and nice and recommended way of solving those kind of issues.

BTW I think I partially reinvented the wheel [here](https://github.com/3mdeb/debian-rootfs-builder) - it would be great if we can align at some point in October. If you have ideas how to do that please let me know.

Reply:
@pietrushnic Using backports repo does seem like a much cleaner solution. I will try that.

I think you did a little bit. At least feature-wise. I'm not yet sure how will the `debos` work in real scenarios we face, though.